algorithms/rotate_matrix.py

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python3
+"""Python solution for CtCI 6th Edition Problem 1.7
+
+Given an image represented by an NxN matrix, where each pixel in the image is
+4 bytes, write a method to rotate the image by 90 degrees. Can you do this in
+place?
+
+Odd sized matrix
+>>> rotate_matrix([['q', 'w', 'e'], ['a', 's', 'd'], ['z', 'x', 'c']])
+[['z', 'a', 'q'], ['x', 's', 'w'], ['c', 'd', 'e']]
+
+Even sized matrix
+>>> rotate_matrix([['1', '2', '3', '4'], ['q', 'w', 'e', 'r'], ['a', 's', 'd', 'f'], ['z', 'x', 'c', 'v']])
+[['z', 'a', 'q', '1'], ['x', 's', 'w', '2'], ['c', 'd', 'e', '3'], ['v', 'f', 'r', '4']]
```

Comment:
What you have is fine, but what I was thinking for readability was more like:
```
[
    ['z', 'a', 'q', '1'],
    ['x', 's', 'w', '2'],
    ['c', 'd', 'e', '3'],
    ['v', 'f', 'r', '4']
]
```
This way you can visualize the matrix easily and hopefully see how a rotation would look more clearly.